packages/backend-function/src/schedule_parser.ts

```diff
@@ -1,159 +1,267 @@
-  /**
-   * Initialize EventBridge rules
-   */
-  constructor(
-    private readonly lambda: NodejsFunction,
-    private readonly timeIntervals: TimeInterval[]
-  ) {
-    this.timeIntervals.forEach((interval, index) => {
-      if (isRate(interval)) {
-        const { value, unit } = parseRate(interval);
+export const convertFunctionSchedulesToRuleSchedules = (
+  lambda: NodejsFunction,
+  schedules: FunctionSchedule[]
+) => {
+  const errors: string[] = [];
+  const ruleSchedules: Schedule[] = [];
 
-        if (value && !isPositiveWholeNumber(value)) {
-          throw new Error(`schedule must be set with a positive whole number`);
-        }
+  schedules.forEach((schedule) => {
+    let hasError = false;
+    if (isTimeInterval(schedule)) {
+      const { value, unit } = parseTimeInterval(schedule);
 
-        if (
-          value &&
-          lambda.timeout &&
-          unit === 'm' &&
-          value * 60 < lambda.timeout.toSeconds()
-        ) {
-          const timeout = lambda.timeout.toSeconds();
-
-          throw new Error(
-            `schedule must be greater than the timeout of ${timeout} ${
-              timeout > 1 ? 'seconds' : 'second'
-            }`
-          );
-        }
-      } else {
-        if (!isValidCron(interval)) {
-          // TODO: Better error messaging here or throw within each part of isValidCron in order to give more concise error messages
-          throw new Error(`schedule cron expression is not valid`);
-        }
+      if (value && !isPositiveWholeNumber(value)) {
+        hasError = true;
+        errors.push(
+          'Function schedule rate must be set with a positive whole number'
+        );
+      } else if (
+        value &&
+        lambda.timeout &&
+        unit === 'm' &&
+        value * 60 < lambda.timeout.toSeconds()
+      ) {
+        hasError = true;
+        const timeout = lambda.timeout.toSeconds();
+        errors.push(
+          `Function schedule rate must be greater than the function timeout of ${timeout} ${
+            timeout > 1 ? 'seconds' : 'second'
+          }`
+        );
       }
+    } else {
+      const cronErrors = validateCron(schedule);
 
-      try {
-        // Lambda name will be prepended to rule id, so only using index here for uniqueness
-        const rule = new Rule(this.lambda, `lambda-schedule${index}`, {
-          schedule: Schedule.cron(translateToCronOptions(interval)),
-        });
-
-        rule.addTarget(new targets.LambdaFunction(this.lambda));
-      } catch (error) {
-        throw new AmplifyUserError(
-          'NodeJSFunctionScheduleInitializationError',
-          {
-            message: 'Failed to instantiate schedule for nodejs function',
-            resolution: 'See the underlying error message for more details.',
-          },
-          error as Error
-        );
+      if (cronErrors.length > 0) {
+        hasError = true;
+        errors.push(...cronErrors);
       }
-    });
+    }
+
+    if (!hasError) {
```

Comment:
nit: I think we can just use `errors.length` as our indicator of whether or not there are errors, and then we can remove the `hasError` boolean

Reply:
You're right, I was initially using this to specifically avoid translating a specific schedule with errors to a rule schedule, which would lead to more errors. With us throwing afterwards this becomes a moot point and `errors.length` will do the job just fine of preventing further errors.